android/src/main/java/com/reactnativehmssdk/HMSView.kt

```diff
@@ -15,57 +15,56 @@ import com.facebook.react.uimanager.events.RCTEventEmitter
 import live.hms.video.media.tracks.HMSVideoTrack
 import live.hms.video.utils.HmsUtilities
 import live.hms.videoview.HMSVideoView
+import live.hms.videoview.VideoViewStateChangeListener
 import org.webrtc.RendererCommon
 
 @SuppressLint("ViewConstructor")
 class HMSView(context: ReactContext) : FrameLayout(context) {
-  private var hmsVideoView: HMSVideoView = HMSVideoView(context)
+  private var hmsVideoView: HMSVideoView? = null
   private var videoTrack: HMSVideoTrack? = null
-  private var scaleTypeApplied: Boolean = false
   private var sdkId: String = "12345"
-  private var currentScaleType: RendererCommon.ScalingType =
-    RendererCommon.ScalingType.SCALE_ASPECT_FILL
   private var disableAutoSimulcastLayerSelect = false
+  private var jsCanApplyStyles = false
 
   init {
     val inflater = getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
     val view = inflater.inflate(R.layout.hms_view, this)
 
     hmsVideoView = view.findViewById(R.id.hmsVideoView)
-    hmsVideoView.setEnableHardwareScaler(false)
-    hmsVideoView.setScalingType(currentScaleType)
-    hmsVideoView.setMirror(false)
-    hmsVideoView.disableAutoSimulcastLayerSelect(disableAutoSimulcastLayerSelect)
+    hmsVideoView?.setEnableHardwareScaler(false)
+
+    hmsVideoView?.setMirror(false)
+    hmsVideoView?.disableAutoSimulcastLayerSelect(disableAutoSimulcastLayerSelect)
+
+    hmsVideoView?.addVideoViewStateChangeListener(object : VideoViewStateChangeListener {
```

Comment:
will the addVideoViewStateChangeListener be applied in init?
Will the listener event be removed?
Are both the callbacks - resolution & frameRendered will be sent to JS side?